### PR TITLE
Implement getScans Rest API

### PR DESCRIPTION
**Pull request by haonliu**
#### Description of changes

(Give an overview. Add a technical description describing your changes.)

#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Added relevant unit test for your changes. (`yarn test`)
- [ ] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [ ] Ran precheckin (`yarn precheckin`)

### CODE REVIEW

File: packages/web-api/src/controllers/batch-scan-result-controller.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        const scanIds = payload.map(request => request.scanId);
+        const responseBody: OnDemandPageScanResultResponse[] = [];
+
+        const scanIdsToQuery: string[] = [];
+        // partition the array into three collections
+        scanIds.forEach(async scanId => {
+            const isRequestMadeTooSoon = await this.isRequestMadeTooSoon(scanId);
+            if (isRequestMadeTooSoon === true) {
+                responseBody.push(this.getTooSoonRequestResponse(scanId));
+            } else if (isRequestMadeTooSoon === undefined) {
```
Comment (devaradhanm) [inline packages/web-api/src/controllers/batch-scan-result-controller.ts]: handle invalid request first before calling getTooSoonRequestResponse. considering the request is invalid if isRequestMadeTooSoon  = undefined looks weird

File: packages/web-api/src/controllers/batch-scan-result-controller.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+import { OnDemandPageScanRunResultProvider } from 'service-library';
+import { ScanResultResponse } from './../api-contracts/scan-result-response';
+
+import { ScanBatchRequest } from './../api-contracts/scan-batch-request';
+import { BaseScanResultController } from './base-scan-result-controller';
+
+@injectable()
+export class BatchScanResultController extends BaseScanResultController {
+    public readonly apiVersion = '1.0';
+    public readonly apiName = 'get-scans';
```
Comment (lamaks) [inline packages/web-api/src/controllers/batch-scan-result-controller.ts]: add web-api- prefix

File: packages/web-api/src/controllers/batch-scan-result-controller.ts [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        @inject(OnDemandPageScanRunResultProvider) protected readonly onDemandPageScanRunResultProvider: OnDemandPageScanRunResultProvider,
+        @inject(GuidGenerator) protected readonly guidGenerator: GuidGenerator,
+        @inject(ServiceConfiguration) protected readonly serviceConfig: ServiceConfiguration,
+        @inject(Logger) protected readonly logger: Logger,
+    ) {
+        super();
+    }
+
+    public async handleRequest(): Promise<void> {
+        const payload = this.tryGetPayload<ScanBatchRequest[]>();
```
Comment (lamaks) [inline packages/web-api/src/controllers/batch-scan-result-controller.ts]: check if payload is empty

File: packages/web-api/src/api-contracts/scan-result-response.ts [hunk 4]
```diff
@@ -5,14 +5,21 @@ export declare type ReportFormat = 'sarif';
 export declare type ScanState = 'unknown' | 'pass' | 'fail';
 export declare type RunState = 'unknown' | 'accepted' | 'queued' | 'running' | 'completed' | 'failed';
 
-export interface ScanResultResponse {
+export interface ScanResultErrorResponse {
```
Comment (lamaks) [inline packages/web-api/src/api-contracts/scan-result-response.ts]: we need to keep client api contract up-to-date, update api-contracts yaml def

File: packages/web-api/src/controllers/base-scan-result-controller.ts [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+export abstract class BaseScanResultController extends ApiController {
+    protected abstract readonly onDemandPageScanRunResultProvider: OnDemandPageScanRunResultProvider;
+    protected abstract readonly guidGenerator: GuidGenerator;
+
+    // tslint:disable-next-line: no-any
+    protected abstract handleInvalidRequest(scanId: string): void;
+
+    protected async isRequestMadeTooSoon(scanId: string): Promise<boolean> {
+        const timeRequested = this.getScanRequestedTime(scanId);
+        const timeCurrent = new Date();
```
[the remark this answers is not in the document]
Comment (haonliu) [reply packages/web-api/src/controllers/base-scan-result-controller.ts]: I'm not following, I'm not using local time here.

**Pull request merged**